client.go

```diff
@@ -247,6 +283,14 @@ func (client *Client) setVariablesToInitializedAPI() error {
 	return nil
 }
 
+func (client *Client) setVariablesToInitializedAPIIfNeeded() error {
```

Comment:
Could you please add more comments and links we should know on this method, so that we can make sure what this method for and why.